src/main/scala/com/microsoft/partnercatalyst/fortis/spark/TransformManager.scala

```diff
@@ -0,0 +1,256 @@
+import com.microsoft.partnercatalyst.fortis.spark.transforms.locations.client.FeatureServiceClient
+import com.microsoft.partnercatalyst.fortis.spark.transforms.sentiment.SentimentDetectorAuth
+import org.apache.spark.SparkContext
+
+import scala.util.Properties
+
+class TransformManager(configManager: ConfigurationManager, featureServiceClient: FeatureServiceClient) extends Serializable with Loggable {
+  private val shouldUpdate: SynchronousQueue[Delta] = new SynchronousQueue[Delta]()
+  private val writeLock: ReentrantLock = new ReentrantLock(true)
+
+  // Do not serialize these values. The transformContext would otherwise contain stale Broadcast
+  // instances, so we rebuild it on recovery from checkpoint.
+  @volatile @transient private var transformContext: TransformContext = null
+  @volatile @transient private var queueClient: QueueClient = null
+
+  def getOrUpdateContext(sparkContext: SparkContext): TransformContext = {
+    ensureInitialized(sparkContext)
+
+    writeLock.lock()
+    try {
+      // Grab delta from hand-off only if it's available *now*
+      val delta = Option(shouldUpdate.poll(0, TimeUnit.SECONDS))
+
+      // Update transform context with delta
+      delta.foreach(updateTransformContext(_, sparkContext))
+    }
+    finally {
+      writeLock.unlock()
+    }
+
+    transformContext
+  }
+
+  private def ensureInitialized(sparkContext: SparkContext): Unit = {
+    if (transformContext == null) {
+      // Blocking init of transform context and queue client used for non-blocking updates.
+      // Initialization has not yet occurred since construction/deserialization.
+      writeLock.lock()
+      try {
+        if (transformContext == null) {
+
+          // Fetch data synchronously
+          val siteSettings = configManager.fetchSiteSettings()
+          val langToWatchlist = configManager.fetchWatchlist()
+          val blacklist = configManager.fetchBlacklist()
+
+          var delta = deltaWithSettings(siteSettings)
+          delta = deltaWithWatchlist(langToWatchlist, delta)
+          delta = deltaWithBlacklist(blacklist, delta)
+
+          // Create transformContext and broadcast
+          updateTransformContext(delta, sparkContext)
+
+          // Start async update listener
+          startQueueClient()
+        }
+      }
+      finally {
+        writeLock.unlock()
+      }
+    }
+  }
+
+  private def startQueueClient(): Unit = {
+    queueClient = new QueueClient(
+      new ConnectionStringBuilder(
+        Properties.envOrElse("FORTIS_SERVICEBUS_NAMESPACE", ""),
+        Properties.envOrElse("FORTIS_SERVICEBUS_CONFIG_QUEUE", ""),
+        Properties.envOrElse("FORTIS_SERVICEBUS_POLICY_NAME", ""),
+        Properties.envOrElse("FORTIS_SERVICEBUS_POLICY_KEY", "")
+      ), ReceiveMode.PeekLock)
+
+    queueClient.registerMessageHandler(
+      new MessageHandler(),
+      new MessageHandlerOptions(
+        1, // Max concurrent calls
+        true,
```

Comment:
Would it be possible to use named arguments here (and for the duration below) so we don't have to guess what this flag refers to? Similarly, named arguments will also make the comment for the integer above redundant.